mlflow/tracking/_tracking_service/client.py

```diff
@@ -607,6 +615,14 @@ def log_artifact(self, run_id, local_path, artifact_path=None):
         else:
             artifact_repo.log_artifact(local_path, artifact_path)
 
+    def download_trace(self, trace_id):
+        artifact_repo = self._get_artifact_repo_for_trace(trace_id)
+        return artifact_repo.download_trace(trace_id)
```

Comment:
This method is only defined on `DatabricksArtifactRepository`. Can we add `download_trace` and `upload_trace` to the base artifact repository class and raise `NotImplemented` by default? Can we also file a follow-up ticket to track implementing `download_trace` and `upload_trace` for all of our other artifact repos?

Reply:
Got it, filed a ticket!